expression/integration_test.go

```diff
@@ -6909,3 +6909,14 @@ func TestIssue30174(t *testing.T) {
 	tk.MustQuery("select * from t2 where c3 in (select c2 from t1);").Check(testkit.Rows())
 	tk.MustQuery("select * from t2 where c2 in (select c2 from t1);").Check(testkit.Rows())
 }
+
+func TestIssue30327(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	result := tk.MustQuery("SELECT cast(TIME'23:59:59.4' as date)")
+	result.Check(testkit.Rows(time.Now().Format("2006-01-02")))
+	tk.MustExec("SET TIMESTAMP=978332400;")
```

Comment:
It might be more readable to convert this to `SET TIMESTAMP=UNIX_TIMESTAMP(t)`.

Reply:
done